apps/system/js/notifications.js

```diff
@@ -403,64 +404,80 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
-    // We need to animate the ambient indicator when the toast
-    // timesout, so we skip updating it here, by passing a skip bool
-    this.addUnreadNotification(detail.id, true);
-
-    var notificationNode = document.createElement('div');
-    notificationNode.classList.add('notification');
-    notificationNode.setAttribute('role', 'link');
-
-    notificationNode.dataset.notificationId = detail.id;
-    notificationNode.dataset.noClear = behavior.noclear ? 'true' : 'false';
-    notificationNode.lang = detail.lang;
-    notificationNode.dataset.predefinedDir = detail.bidi;
-
-    notificationNode.dataset.obsoleteAPI = 'false';
-    if (typeof detail.id === 'string' &&
-        detail.id.indexOf('app-notif-') === 0) {
-      notificationNode.dataset.obsoleteAPI = 'true';
-    }
-    var type = detail.type || 'desktop-notification';
-    notificationNode.dataset.type = type;
-    notificationNode.dataset.manifestURL = manifestURL;
-
-    if (detail.icon) {
-      var icon = document.createElement('img');
-      icon.src = detail.icon;
-      icon.setAttribute('role', 'presentation');
-      notificationNode.appendChild(icon);
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var isNotificationAvailable = this.unreadNotifications.has(manifestURL);
+    if (!isNotificationAvailable) {
+      // We need to animate the ambient indicator when the toast
+      // timesout, so we skip updating it here, by passing a skip bool
+      this.addUnreadNotification(manifestURL, true);
+      var notificationNode = document.createElement('div');
```

Comment:
don't you think that using https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML would be much easier and prettier here instead of using of the document.createElement() function?
The HTML fragment might be defined just like it is done in app_window.js:AppWindow.prototype.view function. 
For insertAdjacentHTML() usage you can check AppWindow.prototype._render function as well as in the docs mentioned above.